libchisel/src/fromwat.rs

```diff
@@ -0,0 +1,96 @@
+extern crate wabt;
+use super::{ModuleCreator, ModuleError};
+use crate::utils::*;
+use parity_wasm::elements::{deserialize_buffer, Module};
+use wabt::Wat2Wasm;
+
+/// Struct on which ModuleCreator is implemented.
+pub struct FromWat<'a> {
+    pub filename: &'a String,
+}
+
+impl<'a> FromWat<'a> {
+    pub fn new(filename: &'a String) -> Result<Self, ()> {
+        Ok(FromWat { filename: filename })
+    }
+}
+
+fn load_file(filename: &str) -> Vec<u8> {
+    use std::fs::File;
+    use std::io::prelude::*;
+    let mut file = File::open(filename).unwrap();
+    let mut buf = Vec::new();
+    file.read_to_end(&mut buf).unwrap();
+    buf
+}
+
+impl From<wabt::Error> for ModuleError {
+    fn from(a: wabt::Error) -> Self {
+        use std::error::Error;
+        ModuleError::Custom(a.description().to_string())
+    }
+}
+
+fn convert_wat(input: &[u8]) -> Result<Module, ModuleError> {
+    // TODO: turn on relocatable(true)?
+    let module = Wat2Wasm::new()
+        .canonicalize_lebs(true)
+        .write_debug_names(true)
+        .convert(&input)?;
+
+    let module = deserialize_buffer::<Module>(module.as_ref())?;
```

Comment:
We can clean this up by doing
```
deserialize_buffer::<Module>(module.as_ref())?
```
as the return, rather than explicitly binding it to the `module` ident and rewrapping it with `Ok`.